agent/acl_endpoint_test.go

```diff
@@ -1341,7 +1361,7 @@ func TestACL_LoginProcedure_HTTP(t *testing.T) {
 				BindName:    "web",
 			}
 
-			req, _ := http.NewRequest("PUT", "/v1/acl/binding-rule?token=root", jsonBody(ruleInput))
```

Comment:
hm, are those breaking changes?

as in, if I'm a practitioner on 1.10.x or 1.11.x and don't URL encode the datacenter, will my existing code still work?

It doesn't seem like it right?

Reply:
`parseDC` should also default the datacenter to the current one:
```
// parseDC is used to parse the ?dc query param
func (s *HTTPHandlers) parseDC(req *http.Request, dc *string) {
	if other := req.URL.Query().Get("dc"); other != "" {
		*dc = other
	} else if *dc == "" {
		*dc = s.agent.config.Datacenter
	}
}
```
I don't think these test assertions should need `&dc=dc1`

Reply:
It is not a breaking change! If no datacenter is provided, it will continue to behave as it previously did. In case someone was passing a query param that referenced a datacenter that was incorrect, that will fail, but that would be desired behavior.

This actually ensures consistent behavior between the read/list requests and the write and delete requests!

Reply:
Updated the tests @rboyer